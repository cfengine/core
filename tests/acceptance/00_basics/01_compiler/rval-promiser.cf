###########################################################
#
# Test rvals in the promiser
#
###########################################################

body common control
{
    inputs => { "../../default.cf.sub" };
    bundlesequence => { default($(this.promise_filename)) };
    version => "1.0";
}

###########################################################

bundle agent test
{
  vars:
      "iter" slist => { "1", "2", "3" };
      canonify(concat("promiser ", canonify("addition"))) string => "";
      canonify("clean") string => "";
      canonify("my name is canonified") string => "";
      concat("iter_canon_", canonify($(iter))) string => "iter = $(iter)";
}

###########################################################

bundle agent check
{
  methods:
      "check"  usebundle => dcs_check_state(test,
                                           "$(this.promise_filename).expected.json",
                                           $(this.promise_filename));
}
