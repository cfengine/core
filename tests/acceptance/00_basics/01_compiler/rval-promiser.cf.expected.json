{
  "clean": "",
  "iter": [
    "1",
    "2",
    "3"
  ],
  "iter_canon_1": "iter = 1",
  "iter_canon_2": "iter = 2",
  "iter_canon_3": "iter = 3",
  "my_name_is_canonified": "",
  "promiser_addition": ""
}
